autopush-common/src/errors.rs

```diff
@@ -29,80 +29,61 @@
 //! You can find some more documentation about this in the `error-chain` crate
 //! online.
 use std::any::Any;
-use std::error;
+//use std::error;
 use std::io;
 use std::num;
 
 use futures::Future;
+use thiserror::Error;
 
-error_chain! {
-    foreign_links {
-        Ws(tungstenite::Error);
-        Io(io::Error);
-        Json(serde_json::Error);
-        Httparse(httparse::Error);
-        MetricError(cadence::MetricError);
-        UuidError(uuid::Error);
-        ParseIntError(num::ParseIntError);
-        ConfigError(config::ConfigError);
-    }
+#[derive(Error, Debug)]
+pub enum Error {
```

Comment:
The big benefit of the separate `ErrorKind` enum and a `struct Error` is it lets us easily capture a `backtrace` on to a single field on the `Error` type.

Though autoconnect currently doesn't utilize the backtraces (in sentry) most of this common code should outlive its current impl., I think? So we should consider supporting backtraces out of the gate now.